README.md

```diff
@@ -352,6 +340,92 @@ Returns one of the following in order of preference:
 In order for this to be most useful, you must configure your controller
 as described in [Change Journaling](#change-journaling) above.
 
+### Testing
+
+If you use RSpec, you can test for journaling behaviors with the
+`journal_event(s)` and `journal_changes_to` matchers. First, make sure to
+require `journaled/rspec` in your spec setup (e.g. `spec/rails_helper.rb`):
+
+```ruby
+require 'journaled/rspec'
+```
+
+#### Checking for specific events
+
+The `journal_event` and `journal_events` matchers allow you to check for one or
+more matching event being journaled:
+
+```ruby
+expect { my_code }.to journal_event(name: 'foo')
+expect { my_code }.to journal_events({ name: 'foo', value: 1 }, { name: 'foo', value: 2 })
+```
+
+This will only perform exact matches on the specified fields (and will not match
+one way or the other against unspecified fields).
+
+The matchers also support negative assertions (in two forms):
+
+```ruby
+expect { my_code }.not_to journal_event
+expect { my_code }.to not_journal_event # supports chaining with `.and`
+```
+
+Several chainable modifiers are also available:
+
+```ruby
+expect { my_code }.to journal_event(name: 'foo')
+  .with_schema_name('my_event_schema')
+  .with_partition_key(user.id)
+  .with_stream_name('my_stream_name')
+  .with_enqueue_opts(run_at: future_time)
+  .with_priority(999)
+```
+
+All of this can be chained together to test for multiple sets of events with
+multiple sets of options:
+
+```ruby
+expect { subject.journal! }
+  .to journal_events({ name: 'event1', value: 300 }, { name: 'event2', value: 200 })
+    .with_priority(10)
+  .and journal_event(name: 'event3', value: 100)
+    .with_priority(20)
+  .and not_journal_event(name: 'other_event')
+```
+
+#### Checking for `Journaled::Changes` declarations
+
+The `journal_changes_to` matcher checks against the list of attributes specified
+on the model. It does not actually test that an event is emitted within a given
+codepath, and is instead intended to guard against accidental regressions that
+may impact external consumers of these events:
+
+```ruby
+it "journals exactly these things or there will be heck to pay" do
+  expect(User).to journal_changes_to(:email, :first_name, :last_name, as: :identity_change)
+end
+```
+
+### Instrumentation
+
+When an event is enqueued, an `ActiveSupport::Notification` titled
+`journaled.event.enqueue` is emitted. Its payload will include the `:event` and
```

Comment:
Open to naming suggestions. I went with `journaled.event.enqueue` thinking that we may eventually want to capture other lifecycle events.